File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maximumprofit=0
    let  smallprice=Infinity
    for(let i = 0;i<prices.length;i++){
        if(prices[i]<smallprice){
            smallprice=prices[i]
        }
        let profit=prices[i]-smallprice
       if(profit>maximumprofit){
           maximumprofit=profit
    }
    }
    return  maximumprofit
    
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    minPrice:=prices[0]
    profit :=0
    for i:=range prices{
        if len(prices)==0{
            return 0
        }
        if minPrice>prices[i]{
            minPrice=prices[i]
        }else if prices[i]-minPrice>profit{
            profit = prices[i]-minPrice
        }
    }
    return profit
}